pom.xml

```diff
@@ -57,10 +57,39 @@
       <artifactId>application-licensing-licensor-api</artifactId>
       <version>${licensing.version}</version>
     </dependency>
+    <!-- Because adal4j is an archived repo, as part of https://github.com/xwikisas/application-office365/issues/26
+      we need to move away from it. Until then, some exclusions are added in order to override old dependencies versions
+      that are brought by it or by transitive dependencies. A comment is added for each dependency that should be
+      removed when this migration is done. -->
     <dependency>
       <groupId>com.microsoft.azure</groupId>
       <artifactId>adal4j</artifactId>
       <version>1.6.7</version>
+      <exclusions>
+        <!-- Exclude json-smart because the declared range is lower than the managed one in some cases. See
+          https://github.com/xwikisas/application-office365/issues/25. -->
+        <exclusion>
+          <groupId>net.minidev</groupId>
+          <artifactId>json-smart</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <!-- This should be removed once #26 is fixed. -->
+    <dependency>
+      <groupId>net.minidev</groupId>
+      <artifactId>json-smart</artifactId>
+      <version>2.4.11</version>
+      <exclusions>
+        <exclusion>
+          <groupId>org.ow2.asm</groupId>
+          <artifactId>asm</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <!-- This should be removed once #26 is fixed. -->
+    <dependency>
+      <groupId>org.ow2.asm</groupId>
+      <artifactId>asm</artifactId>
```

Comment:
You did not specify a version so I'm guessing there is some inherited dependency management that sets the version, but then why do we need to exclude this dependency above, because the dependency management should win (the reason we use a dependency management is because we want a specific version and not the one that comes transitively).

Reply:
After looking at https://github.com/xwikisas/application-office365/pull/27#discussion_r1284388683 , I found this ticket https://bitbucket.org/connect2id/oauth-2.0-sdk-with-openid-connect-extensions/issues/233/shade-json-smart-dependency and it mentions in comments that `asm` is not actually needed, so I'll remove it